tests/std/tests/GH_000690_overaligned_function/test.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+
+#ifndef __cpp_aligned_new
+#error This test is only valid when C++17 over-aligned allocation is supported

```

Comment:
Where's the C++17 dependency? Perhaps I am totally missing something, but all I see is `alignas/alignof` usage which was C++11. There aren't any C++17 checks in the product code, either. Is the issue that when `std::function`'s machinery attempts to dynamically allocate a highly-aligned `_Func_impl_meow`, that's a compiler error in C++14 but not in C++17?

Reply:
Before C++17, global `new`, which allocates over-aligned inner function object, will work as plain new: there's neither `std::align_val_t` overload, nor automatic picking of it in case it exists.

Previously test was written to check only alignment, and implementation did not avoid SFO for re-C++17. Now test can be changed to test SFO in pre-C++17, so I've changed that.